ApeWorX/ape
# feat: set and honor default provider in config

### What I did

fixes: #321

* Add way to set default providers per network in `ape-config.yaml`.
* Causes `test` to be the default default provider for the `development` network (which is what prompted me to make this PR)
* To get the tests to work, I had to force a reload of the ape module in `conftest`.

### How I did it

Use the `ethereum` config.

### How to verify it

Try changing the default provider in the config and see that it works:

```yaml
ethereum:
  development:
    default_provider: geth
```

After the change above, it should start up and use the dev geth:

```bash
ape run <script>
```

Comment out the yaml and see that `test` is the default and run the command again and notice it uses the `test` by  default.

### Checklist
